### DISCUSSION TITLE
Futuresport Passion | Submit project, successor editor wanted

### DISCUSSION THREAD

**Discussion by dc-redeclipse** [redeclipse/base#1193] [Maps and Content Showcase]
**I have given up this project!**
> At this point I will stop continue my work on this map for many reasons.
But if someone will continue working on it feel free to do it!

`Futuresport Passion` is a re-light and ported version of the official team play map 'Futuresport' from Red Eclipse v1.6.
This map was original made by Derek 'JoJo' Stegall and was in Red Eclipse v1.6 official included in map rotation.

![20200120011646](https://user-images.githubusercontent.com/60038236/72691134-699ccf80-3b23-11ea-8d07-701cb25a588e.png)
![20200120011631](https://user-images.githubusercontent.com/60038236/72691135-699ccf80-3b23-11ea-812f-80b9a83daf31.png)
![20200120011609](https://user-images.githubusercontent.com/60038236/72691136-699ccf80-3b23-11ea-876e-10cf870f4228.png)
![20200120011357](https://user-images.githubusercontent.com/60038236/72691140-6acdfc80-3b23-11ea-8b11-e65596a1ef14.png)
(race mode onslaught gauntlet - to finish you had to reach the green ball)
![20200120012010](https://user-images.githubusercontent.com/60038236/72691133-69043900-3b23-11ea-90bb-4bde4c947320.png)

Steps I done for the remade (re-light version):
- fixed blocks
- adjust and scale many block structures for a better gaming experience
- made all way points new
- texture fixed and improved some places to look better and error correction
- reduce light elements and improved the old light settings
- I scale all the key elements up a bit to match the new game physics
- made new home bases because the old ones mismatch with the new re base model in bb in ctf
- add a descriptions and optimize the map size and terrain around
- add extra walls for covering
- add a trigger, start and finish with particles for race mode (works cool in team and with onslaught)
- hdrbright and sunlight improved

Ideal Gameplay:
team & ffa deathmatch / capture the flag / bomberball / duel / onslaught / defend and  control / medieval
team & ffa race / gauntlet / lapped / onslaught / medieval
6-20 players or more

**current version = r672**

Please review it for any errors that this map can be soon back in the official game.

[futuresport-r672.zip](https://github.com/redeclipse/world/files/4239108/futuresport-r672.zip)

> to do:

**Comment by dc-redeclipse** [comment]
**Updated to r510**

    little fixes
    improve mainpost

**Comment by dc-redeclipse** [comment]
I facing with a really wired bug
![20200119110204](https://user-images.githubusercontent.com/60038236/72679238-a0380300-3aad-11ea-81de-c3a97448a24d.png)
if I running normal down the ramp I slowed don each time but there are no blocks on it that cause this!
This happens only by running down not up on any ramp like this!
But why? And why only in 2.0?
In 1.6 is all fine an no slowing down.

**Comment by qreeves** [comment]
Changes to the player size has caused certain corner cases to appear where the hitbox ends up inside geometry too much when stepping up/down. This is because the physics allow a certain amount of overlap to enable the smooth transition up stairs, etc. I'm still yet to fully figure out where the problem is, but I am aware of it.

**Comment by qreeves** [comment]
Here's a rough suggestion on how I would add cover to the map (red lines are occluders like walls, etc):

![image](https://user-images.githubusercontent.com/3297929/72679451-f3847280-3afa-11ea-9160-e87246abd851.png)

**Comment by dc-redeclipse** [comment]
> Changes to the player size has caused certain corner cases to appear where the hitbox ends up inside geometry too much when stepping up/down. This is because the physics allow a certain amount of overlap to enable the smooth transition up stairs, etc. I'm still yet to fully figure out where the problem is, but I am aware of it.

ok I will try to make it more like steps and hoping it helped if I using little alpha blocks.

**Comment by dc-redeclipse** [comment]
> Here's a rough suggestion on how I would add cover to the map (red lines are occluders like walls, etc):
> 
> ![image](https://user-images.githubusercontent.com/3297929/72679451-f3847280-3afa-11ea-9160-e87246abd851.png)

Nice idea! Now I will build some walls to test it. :)

**Comment by dc-redeclipse** [comment]
> > Changes to the player size has caused certain corner cases to appear where the hitbox ends up inside geometry too much when stepping up/down. This is because the physics allow a certain amount of overlap to enable the smooth transition up stairs, etc. I'm still yet to fully figure out where the problem is, but I am aware of it.
> 
> ok I will try to make it more like steps and hoping it helped if I using little alpha blocks.

I got the a solution an maybe the reason for the problem.
The block size was on full length 1/8 with an very high climbing rate.
I reproduce that with an other ramp and also 1/8 blocks got the same effect.
![20200119114933](https://user-images.githubusercontent.com/60038236/72679752-8a790c80-3ab2-11ea-8dc8-c49075d6e369.png)
You can fixed it with full blocks like this.

**Comment by dc-redeclipse** [comment]
**Updated to r527**

    fix lights reduce blending
    fix block size on stairs from 1/8 to full blocks
    add walls for cover and extend the both middle buildings for bigger cove

**Comment by dc-redeclipse** [comment]
![20200119160142](https://user-images.githubusercontent.com/60038236/72683526-6af4da80-3ad8-11ea-9515-8152c0d32f25.png)
I found an error and don't know what it cause.
This little shadow on the edge near the next layer make no sense but its there since 2.0 but why?

**Comment by Calinou** [comment]
> This little shadow on the edge near the next layer make no sense but its there since 2.0 but why?

It's due to screen-space ambient occlusion (SSAO). I don't know if it can be avoided somehow, but it will go away if you enter `/ssao 0` in the console. This only applies to your own client though.

**Comment by SniperGoth** [comment]
Smoothen those edges, as AO appears more around angled surfaces/corners, and that should go away.

**Comment by dc-redeclipse** [comment]
> > This little shadow on the edge near the next layer make no sense but its there since 2.0 but why?
> 
> It's due to screen-space ambient occlusion (SSAO). I don't know if it can be avoided somehow, but it will go away if you enter `/ssao 0` in the console. This only applies to your own client though.

Yes thank you I fixed it with extra blocks to make it more flat and it worked well :)

**Comment by dc-redeclipse** [comment]
**Updated to r581**

    fix environment lights and skybox
    improve cover walls in the middle section
    texture fix
    add start and finish for a little race mode. used trigger particles for that, race mode on this map makes a lot of fun with team or ffa with gauntlet or and onslaught

**Comment by dc-redeclipse** [comment]
**Updated to r595**

    improve hdrbright and sunlight and also fog (thx unixfreak for your help)
    improve cover walls in the middle section
    some texture fixes and improvements

**Comment by qreeves** [comment]
The walls added for cover probably need to be larger, but I think we also need to prevent people from standing on top of them somehow (death material spikes? I dunno). The main point is to reduce the amount of places in the map where there is a clear line of sight for the rifle.

**Comment by dc-redeclipse** [comment]
> The walls added for cover probably need to be larger, but I think we also need to prevent people from standing on top of them somehow (death material spikes? I dunno). The main point is to reduce the amount of places in the map where there is a clear line of sight for the rifle.

I like to use death block with a trigger for a lightning strike, like this on all higher walls:
![20200120135526](https://user-images.githubusercontent.com/60038236/72735246-b4f2c480-3b9b-11ea-985e-c467c20d2c1a.png)
(One wall with it is including the last update [r631] for testing.)

What do you mean? good idea or bad solution?

**Comment by dc-redeclipse** [comment]
**Updated to r631**

    improve walls and add some more
    set death blocks and trigger with particles for testing on one middle wall
    some texture fixes and improvements
    add names for the id's

**Comment by dc-redeclipse** [comment]
**Updated to r672**

    bigger walls ect...

> At this point I will stop continue my work on this map for many reasons.
But if someone will continue working on it feel free to do it!
